test/unit/ClusterServiceTest.js

```diff
@@ -0,0 +1,404 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const chai = require('chai');
+chai.use(require('chai-as-promised'));
+const expect = chai.expect;
+const sinon = require('sinon');
+const sandbox = sinon.createSandbox();
+
+const { ClientConfigImpl } = require('../../lib/config/Config');
+const { MemberInfo } = require('../../lib/core/MemberInfo');
+const { ClusterService } = require('../../lib/invocation/ClusterService');
+const { AddressImpl } = require('../../');
+const { UuidUtil } = require('../../lib/util/UuidUtil');
+const { DefaultLogger } = require('../../lib/logging/DefaultLogger');
+const { ConnectionRegistryImpl } = require('../../lib/network/ConnectionRegistry');
+
+describe('ClientClusterServiceImplTest', function () {
+    let loggerStub;
+    let clusterFailoverServiceStub;
+    let connectionRegistryStub;
+    let clientConfig;
+    let clusterService;
+    const INITIAL_MEMBER_LIST_VERSION = -1;
+
+    before(async function () {
+        clientConfig = new ClientConfigImpl();
+    });
+
+    beforeEach(async function () {
+        loggerStub = sandbox.stub(DefaultLogger.prototype);
+        clusterFailoverServiceStub = {
+            current: () => {
+                return {};
+            }
+        };
+        connectionRegistryStub = sandbox.stub(ConnectionRegistryImpl.prototype);
+        clusterService = new ClusterService(
+            clientConfig,
+            loggerStub,
+            clusterFailoverServiceStub
+        );
+        clusterService.translateToAddressProvider.refresh = () => {
+            return Promise.resolve();
+        };
+    });
+
+    afterEach(async function () {
+        sandbox.restore();
+    });
+
+    function createMember({host, port, uuid, liteMember}) {
+        const address = new AddressImpl(host, port);
+        const memberUUID = uuid ? uuid : UuidUtil.generate();
+        return new MemberInfo(address, memberUUID, new Map(), liteMember, null, false, null);
```

Comment:
for version, instead of null, we can use an empty string. 

```suggestion
        return new MemberInfo(address, memberUUID, new Map(), liteMember, '', false, null);
```

Reply:
Fixed